### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by brian6932** [digitalguy99/pwsh-syntax-highlighting#4]
When the module is imported, there seems to add a good amount of input lag after typing spaces, it freezes actions done after it for a bit too long, maybe there's a way to reduce this, some way to do this asynchronously?

**Comment by digitalguy99** [comment]
I don't get much of an input lag. Maybe you can attach a screen recording so I might be able to see what the problem is.

**Comment by brian6932** [comment]
I don't think I'd be able to really record this well, but I could give it a try when I have some time, maybe with KeyViz or something. I did notice some other stuff though, when `cmd_validator.ps1` is used, doing things like `ctrl + a` then typing to override the written text, doesn't work properly.

**Comment by digitalguy99** [comment]
Has it always been like this? Or did this issue just appear when I fixed the problem with the square brackets?

**Comment by brian6932** [comment]
Yes it was like this before the square brackets issue was fixed as well, just didn't report at the time.

**Comment by digitalguy99** [comment]
Hi, @brian6932. Sorry it took so long, can you try again now?

**Comment by brian6932** [comment]
This module still causes some weird behaviors with PSReadLine. For example, when it's imported, and I press <kbd>CTRL</kbd>+<kbd>A</kbd> to highlight all text, then keep typing to overwrite it, this module just skips the cursor to the end of the line instead of ovewriting the text. While this should be it's own issue, I felt like mentioning it here, since this completely breaks my chances of using it.

**Comment by digitalguy99** [comment]
I see, thanks for the detailed explanation of the issue. I will take a look into it some more.

**Comment by digitalguy99** [comment]
@brian6932, I've pushed some new code and tested it on both Conhost and Windows Terminal. Mind giving it another try?

**Comment by brian6932** [comment]
Ok <kbd>CTRL</kbd>+<kbd>A</kbd> issue seems to be fixed now. While typing, the cursor does occasionally still move to the beginning and then jump back to end though.

**Comment by digitalguy99** [comment]
Can you show me with a screen recording, @brian6932 ? It doesn't seem to occur to me at all.

**Comment by digitalguy99** [comment]
I have tried a different implementation. It's not perfect but see if this solves your problem, @brian6932:
1. Uninstall the current module you have installed.
2. Run this:
    ```pwsh
    ($content = irm 'https://raw.githubusercontent.com/digitalguy99/pwsh-syntax-highlighting/refs/heads/feature/xplat-compatible/syntax-highlighting.psm1') | Out-File ($path = "$HOME/syntax-highlighting.ps1"); iex $content; Add-Content $profile ". '$path'"
    ```

**Comment by brian6932** [comment]
So this branch doesn't seem to cause any jumping cursors anymore, and I'd consider these 2 issues fixed.
